src/types/utils.js

```diff
@@ -9,3 +9,10 @@
  * https://flow.org/try/#0C4TwDgpgBAcg9sAanAlgEygXigKClAHygDsBXAWwCMIAnPQqAZ2BpWIHN6jK44AbCAENiXEqT59RAQRo1BIADzCQAPlEB5SgCsIAY2ABuHDgBmpYvpRxiUMxYUAVAFywEydCoAUAN0F9SEC4OAJRBUADe9DQQwKQ0Nr7+EEYAvsZ2up4AjMFGGZ4ARMysHAW5puaZ4ba8LgDklII0dVAp5flkEu2VngDaWQA0UABMALrdFp7maBAmbBBoE5nlQA
  */
 export type NotVoidOrNull = number | string | boolean | Array<any> | Object;
+
+export type ExtractReturnType = <V>((...args: any[]) => V) => V;
```

Comment:
I think (not sure) that you can even leave out `...args: any[]`: 
```js
 export type ExtractReturnType = <V>(() => V) => V;
```

Reply:
Just tried it. It doesn't match the signature, so it doesn't get applied.